staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go

```diff
@@ -150,6 +153,41 @@ func SetFeatureGatesDuringTest(tb TB, gate featuregate.FeatureGate, features Fea
 	})
 }
 
+func newLogger(tb TB) klog.Logger {
+	logger, _ := ktesting.NewTestContext(testingT{tb})
+	return logger
+}
+
+// testingT is a bridge between what this package traditionally has required (Logf) and
+// what ktesting needs (Log).
+type testingT struct {
+	TB
+}
+
+func (tb testingT) Log(args ...any) {
+	// testing.T.Log emulates fmt.Println, not fmt.Print, i.e. it always
+	// adds spaces between arguments.
+	tb.Logf("%s", fmt.Sprintln(args...))
+}
+
+func setFromMap(logger klog.Logger, gate featuregate.FeatureGate, m map[string]bool) error {
+	if gateWithLogger, ok := gate.(featuregate.MutableFeatureGateWithLogger); ok {
```

Comment:
It's debatable whether this extra check is necessary. Already before it was assumed that the `featuregate.FeatureGate` implements also `featuregate.MutableFeatureGate`. I added the check because theoretically the implementation could be something other than `featureGate` and I didn't want to break such usage.